demo/utils/services/dataProvider/sim-tracks-data-provider.ts

```diff
@@ -7,43 +7,91 @@ import { AcNotification } from '../../../../src/angular-cesium/models/ac-notific
-			socket.on('birds', (data) => {
-				data.forEach(
-					(acNotification) => {
-						if (acNotification.action === 'ADD_OR_UPDATE') {
-							acNotification.actionType = ActionType.ADD_UPDATE;
-							acNotification.entity = new AcEntity(this.convertToCesiumObj(acNotification.entity));
-						}
-						else if (acNotification.action === 'DELETE') {
-							acNotification.actionType = ActionType.DELETE;
-						}
-						observer.next(acNotification);
-					});
-			});
-		});
-		
-		return simTracks$;
-	}
-	
-	convertToCesiumObj(entity): any {
-		const fixedHeading = entity.heading - (Math.PI / 2);
-		const pitch = Cesium.Math.toRadians(0.0);
-		const roll = Cesium.Math.toRadians(0.0);
-		
-		entity.scale = entity.id === 1 ? 0.3 : 0.15;
-		entity.alt = Math.round(entity.position.altitude);
-		entity.position = Cesium.Cartesian3.fromDegrees(entity.position.long, entity.position.lat, entity.alt);
-		entity.futurePosition =
-			Cesium.Cartesian3.fromDegrees(entity.futurePosition.long, entity.futurePosition.lat, entity.futurePosition.altitude);
-		const hpr = new Cesium.HeadingPitchRoll(fixedHeading, pitch, roll);
-		entity.orientation = Cesium.Transforms.headingPitchRollQuaternion(entity.position, hpr);
-		return entity;
-	}
-}
\ No newline at end of file
+
+  constructor(private webSocketSupplier: WebSocketSupplier) {
+  }
+
+  get(): Observable<AcNotification> {
+    const socket = this.webSocketSupplier.get();
+    const simTracks$ = Observable.create((observer) => {
+      socket.on('birds', (data) => {
+        data.forEach(
+          (acNotification) => {
+            if (acNotification.action === 'ADD_OR_UPDATE') {
+              acNotification.actionType = ActionType.ADD_UPDATE;
+              acNotification.entity = new AcEntity(this.convertToCesiumObj(acNotification.entity));
+            }
+            else if (acNotification.action === 'DELETE') {
+              acNotification.actionType = ActionType.DELETE;
+            }
+            observer.next(acNotification);
+          });
+      });
+    });
+
+    return simTracks$;
+  }
+
+  convertToCesiumObj(entity): any {
+    const fixedHeading = entity.heading - (Math.PI / 2);
+    const pitch = Cesium.Math.toRadians(0.0);
+    const roll = Cesium.Math.toRadians(0.0);
+
+    entity.scale = entity.id === 1 ? 0.3 : 0.15;
+    entity.alt = Math.round(entity.position.altitude);
+    entity.array = this.getArrayBySize(entity, Math.round(Math.random() * 3));
+    entity.position = Cesium.Cartesian3.fromDegrees(entity.position.long, entity.position.lat, entity.alt);
+    entity.futurePosition =
+      Cesium.Cartesian3.fromDegrees(entity.futurePosition.long, entity.futurePosition.lat, entity.futurePosition.altitude);
+    const hpr = new Cesium.HeadingPitchRoll(fixedHeading, pitch, roll);
+    entity.orientation = Cesium.Transforms.headingPitchRollQuaternion(entity.position, hpr);
+    return entity;
+  }
+
+  getArrayBySize(entity, size) {
```

Comment:
we get the innter arrray for data ffrom the server?

Reply:
just some mock data